execution/executor/src/block_executor/mod.rs

```diff
@@ -232,6 +232,16 @@ where
                         )))
                     });
 
+                    // In case block executor has a cache manager, we need to mark it as ready for
+                    // execution. If for some reason this fails, return an error.
+                    if let Some(module_cache_manager) = self.block_executor.module_cache_manager() {
+                        if !module_cache_manager.mark_ready(&parent_block_id, block_id) {
+                            return Err(ExecutorError::internal_err(
+                                "Unable to mark module cache manager as ready",
+                            ));
+                        }
+                    }
```

Comment:
ideally, this should be inside, by_transaction_execution should accept Some(parent_block_id)

but this PR is too big already, so we can clean that up afterwards

Reply:
I am not sure if it is better to call this here, or propagate it through the calls. I think we can encode this information in `CachedStateView`. But can do this as a follow up?

Reply:
yeah definitelly in the separate PR :)

But from the layers of abstraction - thinking about what are other implementaitons of VMBlockExecutor is a way to reason about things. 
We have (in this pr: #15152) second implementation of that interface - NativeParallelUncoordinatedBlockExecutor - and it doesn't have anything to do with module_cache_manager. So having module_cache_manager here - leaks an implementation detail, as it only applies to AptosVMBlockExecutor implementation.

Reply:
Fixed now by propagating hashvalues to block aptos vm